src/main/java/com/amazon/ion/impl/IonReaderTextSystemX.java

```diff
@@ -1229,7 +1228,11 @@ protected Macro loadMacro() {
                 throw new IonException("E-expressions must begin with an address.");
             }
             Macro macro;
-            if (encodingContext == null || ((macro = encodingContext.getMacroTable().get(address)) == null)) {
+            if (encodingContext == null) {
+                if ((macro = SystemMacro.get(address)) == null) {
```

Comment:
It compiles to the same bytecode, but I think it's more readable in this case if you split the assignment onto a separate line.
```suggestion
                macro = SystemMacro.get(address);
                if (macro == null) {
```

Reply:
Done